### ISSUE TITLE
Map upgrade parameters aren't self-explanatory

### ISSUE THREAD

**Issue by deniz1a** [OpenRA/OpenRA#9306]
This map works in release and playtest but is not supported in bleed:
http://resource.openra.net/maps/4219

Utility crashes when I try to upgrade map to bleed. Check-yaml gives the following errors:

```
Testing map: GreatTreeDivide
OpenRA.Utility(1,1): Error: Missing Type: TargetableBuildingInfo
OpenRA.Utility(1,1): Error: Missing Type: TargetableBuildingInfo
OpenRA.Utility(1,1): Error: Actor CustomHeliHusk is not defined by any rule.
OpenRA.Utility(1,1): Error: Actor CustomMammothHusk is not defined by any rule.
OpenRA.Utility(1,1): Error: Actor CustomHarvesterHusk is not defined by any rule.
```
##

**Comment by Mailaender** [comment]
We need the stacktrace of the upgrade utility command, not the check yaml output of the not yet upgraded map.

**Comment by LipkeGu** [comment]
--map -upgrade isnt supported in bleed :/ the utility is crashing on bleed because of {DEV_VERSION}

**Comment by Mailaender** [comment]
It is a date, not a version!

**Comment by LipkeGu** [comment]
thats wrong the description says Version...

```
[Desc("MAP", "CURRENTENGINE", "Upgrade map rules to the latest engine version.")]
```

**Comment by Phrohdoh** [comment]
@Mailaender you've provided no reason as to why you've closed this ticket (here, not on IRC where I had to *ask*).

@deniz1a please provide complete information without being asked, you've been part of this project long enough to know how people operate here (and how busy we are already).

@LipkeGu Yes technically is has the word `version`, but we use dates as versions, you know this.
`{DEV_VERSION}` is obviously bogus input.

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/wiki/Upgrading-mods-to-a-new-release explains it, but I agree that inline description could reflect this better.

**Comment by deniz1a** [comment]
This is the error of --upgrade-map:
````
PS D:\Dev\openra_dev\OpenRA> .\openra.utility ra --upgrade-map C:\Users\Deniz\Documents\OpenRA\maps\ra\GreatTreeDivide.o
ramap
Error: Utility application crashed. See utility.log for details

Unhandled Exception: System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at OpenRA.Mods.Common.UtilityCommands.UpgradeMapCommand.Run(ModData modData, String[] args)
   at OpenRA.Utility.Program.Main(String[] args) in d:\Dev\openra_dev\OpenRA\OpenRA.Utility\Program.cs:line 62
````

**Comment by deniz1a** [comment]
The --upgrade-map utility command doesn't work. This is not related to the map. ````openra.utility ra --upgrade-map```` crashes with the same error too.

**Comment by penev92** [comment]
Yes. It expects another parameter. In this case - the engine version the map was last updated to.
I believe we're done here?

**Comment by deniz1a** [comment]
Yes it worked when I gave the current engine version parameter, thanks. But it has to be just the date part and should not include release- or playtest-.

**Comment by Phrohdoh** [comment]
@deniz1a please read the builtin help for commands.

**Comment by Unit158** [comment]
This brings up a commonly found issue that's on my long term wishlist. I might use my extra free time on christmas to fix it.

**Comment by LavenderMoon** [comment]
What is that issue?

**Comment by LipkeGu** [comment]
The Error check and check for the arguments... is one of them ...

**Comment by abcdefg30** [comment]
I think the error messages are good (and much better than a cryptic exception).

**Comment by LipkeGu** [comment]
maybe i could add the syntax to "not enough arguments ... because no one see the syntax on the first view @Mailaender  is this OK?

**Comment by Mailaender** [comment]
Maybe you can solve that in a generic way for every command as the parameters are already listed in the [Desc] part.

**Comment by pchote** [comment]
Fixed by #14964.
